optuna_dashboard/ts/trialFilter.ts

```diff
@@ -116,6 +116,32 @@ export const useFilteredTrials = (
     })
   }, [study?.trials, targets, filterComplete, filterPruned])
 
+export const useFilteredTrialsFromStudies = (
+  studies: StudyDetail[],
+  targets: Target[],
+  filterComplete: boolean,
+  filterPruned: boolean
+): Trial[][] =>
+  useMemo<Trial[][]>(() => {
+    return studies.map((study) => {
+      if (study === null) {
+        return []
+      }
+      return study.trials.filter((t) => {
+        if (t.state !== "Complete" && t.state !== "Pruned") {
+          return false
+        }
+        if (t.state === "Complete" && filterComplete) {
+          return false
+        }
+        if (t.state === "Pruned" && filterPruned) {
+          return false
+        }
+        return targets.every((target) => target.getTargetValue(t) !== null)
+      })
+    })
+  }, [studies, targets, filterComplete, filterPruned])
```

Comment:
Do you think we can refactor these lines like below?

```suggestion
  studies.map(s => useFilteredTrials(s, targets, filterComplete, filterPruned))
```

Reply:
If the number of studies becomes different, we call a different number of `useMemo`, and it outputs the following warnings and errors.

```
react_devtools_backend.js:2655 Warning: React has detected a change in the order of Hooks called by GraphHistory. This will lead to bugs and errors if not fixed. For more information, read the Rules of Hooks: https://reactjs.org/link/rules-of-hooks
```

I created `getFilteredTrials` to unify the impementation in 
[c7c227a](https://github.com/optuna/optuna-dashboard/pull/380/commits/c7c227a24a7dd40c6c00f8ae62bdb3675b4a5f34). How about this solution?